bin/comment_with_dependency_diff

```diff
@@ -0,0 +1,153 @@
+#!/bin/bash
+
+set -euo pipefail
+
+MODULE="${1:-}"
+CONFIGURATION="${2:-}"
+
+if [ -z "$MODULE" ] || [ -z "$CONFIGURATION" ]; then
+  echo "Not enough arguments provided. Usage: ./comment_with_dependency_diff <MODULE> <CONFIGURATION>"
+  exit 1
+fi
+
+OWNER=$(basename "$(dirname "${BUILDKITE_PULL_REQUEST_REPO}")")
+REPO=$(basename "${BUILDKITE_PULL_REQUEST_REPO%.git}")
+
+DIFF_DEPENDENCIES_FOLDER="./build/reports/diff"
+BASE_BRANCH_DEPENDENCIES_FILE="$DIFF_DEPENDENCIES_FOLDER/base_branch_dependencies.txt"
+HEAD_BRANCH_DEPENDENCIES_FILE="$DIFF_DEPENDENCIES_FOLDER/head_branch_dependencies.txt"
+BASE_BRANCH_BUILD_ENV_FILE="$DIFF_DEPENDENCIES_FOLDER/base_branch_build_env.txt"
+HEAD_BRANCH_BUILD_ENV_FILE="$DIFF_DEPENDENCIES_FOLDER/head_branch_build_env.txt"
+DIFF_DEPENDENCIES_FILE="$DIFF_DEPENDENCIES_FOLDER/diff_dependencies.txt"
+DIFF_BUILD_ENV_FILE="$DIFF_DEPENDENCIES_FOLDER/diff_build_env.txt"
+DEPENDENCY_TREE_VERSION="1.2.1"
+
+PR_NUMBER=$BUILDKITE_PULL_REQUEST
+PR_URL="https://api.github.com/repos/$OWNER/$REPO/pulls/$PR_NUMBER"
+
+git config --global user.email "mobile@automattic.com"
+git config --global user.name "Dependency Tree Diff Tool"
+
+echo "--> Starting the check"
+echo "--> Fetching the base branch from $PR_URL"
+githubResponse="$(curl "$PR_URL" -H "Authorization: token $GITHUB_TOKEN")"
+baseBranch=$(echo "$githubResponse" | tr '\r\n' ' ' | jq '.base.ref' | tr -d '"')
+echo "--> Base branch is $baseBranch"
```

Comment:
🤔  Actually, is the purpose of this whole API call only to extract the name of the base (aka target) branch of the PR?

If so, why not just use `[BUILDKITE_PULL_REQUEST_BASE_BRANCH](https://buildkite.com/docs/pipelines/environment-variables#BUILDKITE_PULL_REQUEST_BASE_BRANCH)` directly?

_(I think I mentioned this a while ago when I reviewed the first iterations of this script back when it was added to the repositories themselves, and iirc the argument was mostly due to legacy with this script previously being also used in other CI, maybe while we were still in the middle of transitioning to Buildkite…? So maybe that's where it comes from?)_

Reply:
ah yes, thank you. It simplified things a lot!